### DISCUSSION TITLE
QZ integration

### DISCUSSION THREAD

**Discussion by cagnulein** [xdream-biking/trixter-xdream-bike#94] [Ideas]
Hi @drmason789 , do you want to start an integration of this project with QZ?
Let me know in case!

**Comment by drmason789** [comment]
I've already written a C++ class that takes in characters the bike produces and calculates CSCS data. There's not actually much to it. 

Much of this repo is not required to convert input to steering and CSCS data. This repo has code for testing, for emulating the bike to drive the software, for smoothing data and calculating useless details like extremely slow back-pedalling.

I was planning on adding an additional bike to Qdomyos that uses it. I've not yet worked out how to locate the USB serial port using Qt/C++. I could just push some initial code and maybe someone will help. :-)

**Reply by cagnulein** [reply]
if you want here you can find a branch where I already handled a serial port for the Wahoo KICKR bike https://github.com/cagnulein/qdomyos-zwift/tree/dircon-client-
You can branch this one maybe?

**Reply by cagnulein** [reply]
you can use this file as reference https://github.com/cagnulein/qdomyos-zwift/blob/489dd20cc71ae998ee87f500f921ae3c63fac6a5/src/serialdircon.cpp

**Comment by drmason789** [comment]
I've created #95 for this. I've done some initial development here and pushed it. 

I will look at integrating it into a branch off your dircon-client branch.

**Reply by cagnulein** [reply]
great!
